src/components/Permissions/DataList.spec.jsx

```diff
@@ -17,13 +37,152 @@ jest.mock('components/PageTitle', () => {
+          permissions: {
+            files: {
+              type: 'io.cozy.files',
+              description: 'Required to access the files'
+            }
+          },
+          slug: 'settings',
+          name: 'Settings'
+        },
+        {
+          permissions: {},
+          slug: 'store',
+          name: 'Store'
+        }
+      ]
+    }
+    const queryResultKonnectors = {
+      fetchStatus: 'loaded',
+      data: [
+        {
+          permissions: {
+            apps: {
+              type: 'io.cozy.apps',
+              description:
+                'Required by the cozy-bar to display the icons of the apps',
+              verbs: ['GET']
+            }
+          },
+          slug: 'alan',
+          name: 'Alan'
+        }
+      ]
+    }
+    isQueryLoading.mockReturnValue(true)
+    hasQueryBeenLoaded.mockReturnValue(true)
+    Q.mockReturnValue({ getById: () => 'kfrf' })
+    // useQuery.mockReturnValue(permissionsToDisplay)
+    useQuery.mockReturnValueOnce(queryResultApps)
+    useQuery.mockReturnValueOnce(queryResultKonnectors)
+    useQuery.mockReturnValueOnce(queryResultApps)
+    useQuery.mockReturnValueOnce(queryResultKonnectors)
+  })
+
+  it('should render a spinner when query is loading and has not been loaded', () => {
+    isQueryLoading.mockReturnValue(true)
+    hasQueryBeenLoaded.mockReturnValue(false)
+    const { queryByTestId } = render(<DataList />)
+    expect(queryByTestId('Spinner')).toBeTruthy()
+  })
+
   it('should display DataList title', () => {
-    const { getByText } = render(
-      <AppLike>
-        <DataList />
-      </AppLike>
+    const { queryByText } = render(<DataList />)
+    expect(queryByText('Permissions.data')).toBeTruthy
+  })
+
+  it('should display sorted permissions names when query is not loading', () => {
+    isQueryLoading.mockReturnValue(false)
+    const { queryAllByTestId, container } = render(<DataList />)
+    expect(container).toMatchSnapshot()
+    expect(queryAllByTestId('ListItemText')[0]).toHaveAttribute(
+      'data-primary',
+      'CozyClient.Permissions.io.cozy.apps'
     )
-    expect(getByText('Data')).toBeTruthy()
+    expect(queryAllByTestId('ListItemText')[1]).toHaveAttribute(
+      'data-primary',
+      'CozyClient.Permissions.io.cozy.files'
+    )
+    expect(queryAllByTestId('ListItemText')[2]).toHaveAttribute(
+      'data-primary',
+      'CozyClient.Permissions.io.cozy.files.*'
+    )
+  })
+
+  it('should display permissions icons when query is not loading', () => {
+    isQueryLoading.mockReturnValue(false)
+    const { queryAllByTestId, container } = render(<DataList />)
+    expect(container).toMatchSnapshot()
+    expect(queryAllByTestId('Icon')).toBeTruthy()
+  })
+
+  it('should render Permissions.failedRequest when query status is failed', () => {
+    useQuery.mockReturnValue({ fetchStatus: 'failed' })
+    const { queryAllByText } = render(<DataList />)
+    expect(queryAllByText('Permissions.failedRequest')).toBeTruthy()
```

Comment:
why are you using a matcher to get all, if you need only one?

Reply:
when I don't do `queryAllByText`Ihave this error 
```
expect(received).toBeTruthy()
Received: null
```

Reply:
queryByText

Reply:
same that [here](https://github.com/cozy/cozy-settings/pull/440#discussion_r960742936)